pkg/prometheus/operator.go

```diff
@@ -1799,6 +1799,30 @@ func (c *Operator) selectServiceMonitors(ctx context.Context, p *monitoringv1.Pr
 				break
 			}
 
+			if endpoint.Interval != "" && endpoint.ScrapeTimeout != "" {
+				if err = validateScrapeIntervalAndTimeout(endpoint.Interval, endpoint.ScrapeTimeout); err != nil {
+					break
+				}
+			}
+
+			if endpoint.Interval == "" && endpoint.ScrapeTimeout != "" {
+				if err = validateScrapeIntervalAndTimeout(p.Spec.ScrapeInterval, endpoint.ScrapeTimeout); err != nil {
+					break
+				}
+			}
+
+			if endpoint.Interval != "" && endpoint.ScrapeTimeout == "" {
+				if err = validateScrapeIntervalAndTimeout(endpoint.Interval, p.Spec.ScrapeTimeout); err != nil {
+					break
+				}
+			}
+
+			if endpoint.Interval == "" && endpoint.ScrapeTimeout != "" {
+				if err = validateScrapeIntervalAndTimeout(p.Spec.ScrapeInterval, endpoint.ScrapeTimeout); err != nil {
+					break
+				}
+			}
```

Comment:
I would encapsulate the logic into `validateScrapeIntervalAndTimeout()` so that we don't have to repeat the same tests in different places:

```
func validateScrapeIntervalAndTimeout(p *v1.Prometheus, scrapeInterval, scrapeTimeout string) error {
    ...
}
```